src/main/java/com/gamepedia/ftb/oredictdumper/misc/OreDictOutputFormat.java

```diff
@@ -0,0 +1,72 @@
+package com.gamepedia.ftb.oredictdumper.misc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public abstract class OreDictOutputFormat {
+    private final String extension;
+
+    OreDictOutputFormat(String extension) {
+        this.extension = extension;
+    }
+
+    @Nonnull
+    public String getFileExtension() {
+        return extension;
+    }
+
+    @Nonnull
+    public abstract String parseEntries(List<OreDictEntry> entries);
+
+    public static class JSONOutputFormat extends OreDictOutputFormat {
+        public JSONOutputFormat() {
+            super("json");
+        }
+
+        @Nonnull
+        @Override
+        public String parseEntries(List<OreDictEntry> entries) {
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            return gson.toJson(entries);
+        }
+    }
+
+    public static class CSVOutputFormat extends OreDictOutputFormat {
+        public CSVOutputFormat() {
+            super("csv");
+        }
+
+        @Nonnull
+        @Override
+        public String parseEntries(List<OreDictEntry> entries) {
+            StringBuilder builder = new StringBuilder();
+            builder.append("Tag,ItemName,Metadata,ModID\n");
+            for (OreDictEntry entry : entries) {
+                builder.append(String.format("%s,%s,%s,%s\n", entry.tagName, entry.displayName, entry.metadata, entry.modID));
```

Comment:
While it's being rewritten, the fields in OreDictEntry should be made final and private, and this stuff should be replaced by getters.